### PR TITLE
Add custom attributes support

### LINKED ISSUE #19: Add support for custom user parameters
**Is your feature request related to a problem? Please describe.**
There's no support for custom user parameters in user builder.

**Describe the solution you'd like**
Add optional parameter e.g. `Map<String, String> custom` to `init` method. It can call corresponding builder methods for Android/iOS sdk:

```java
builder.custom("param", "value")
```

```swift
user.custom = ["param": "value"]
```

**Describe alternatives you've considered**
Nothing I can think of.

**Additional context**
No.

### PR DESCRIPTION
**Pull request by ookami-kb**
### Requirements

- [x] I have added test coverage for new or changed functionality
- [x] I have followed the repository's [pull request submission guidelines](../blob/master/CONTRIBUTING.md#submitting-pull-requests)
- [x] I have validated my changes against all supported platform versions

### Related issues

Fixes #19 

### Describe the solution you've provided

Added optional `custom` argument to `init` method.

### Describe alternatives you've considered

No.

### Additional context

No.

### How to test?

Pass `Map<String, dynamic> custom` argument to `init` method (e.g. `{'custom': 'value'}`). Check that the corresponding user in LD has this custom attribute set.

### Future works

1. Maybe we should enforce validation for custom parameters. For Android, it checks value types and calls overloaded `custom` methods; if values are of the wrong type it ignores the value at all. For iOS, it just passes the argument as `[String: Any]?` to user data.
2. In Android, we always have user builder, so we pass custom attributes always as well. In iOS, we only create user object if the user key is passed, so custom attributes are only set for an authenticated user. Maybe we should unify this behavior, either by ignoring custom attributes in Android for an anonymous user or by always creating user object in iOS (but it will require creating user key as well).

### CODE REVIEW

**Comment by andre-paraense** [comment]
Awesome, @ookami-kb . Hero! I will definitely find sometime to review your PR this week so we can have a new version as soon as possible! ;)

File: test/launchdarkly_flutter_test.dart [hunk 1]
```diff
@@ -85,10 +85,16 @@ void main() {
     expect(await launchdarklyFlutter.init('MOBILE_KEY', null), true);
   });
 
-  test('init with all arguments', () async {
+  test('init with mobile key and user', () async {
     expect(await launchdarklyFlutter.init('MOBILE_KEY', 'USER_ID'), true);
   });
 
+  test('init with all arguments', () async {
```
Comment (andre-paraense) [inline test/launchdarkly_flutter_test.dart:92]: Can we have a few more tests with edge cases, please?
- custom with other types, not only String, as in this case. For instance, also:  Number, Boolean and also `null`.

**Review by andre-paraense — CHANGES_REQUESTED** [review]

**Comment by andre-paraense** [comment]
@ookami-kb if you can update the example app in the repository, it would be a huge bonus for the community. It is always easier for people to adopt the new feature you have contributed if the example app has and example of its usage ;)

Even better if you also update the Readme in the example folder.

File: test/launchdarkly_flutter_test.dart (hunk 1 above)
Comment (ookami-kb) [reply test/launchdarkly_flutter_test.dart:92]: I've added these types into the test. As for me, it makes sense to have them in one test: `null` case is covered by other cases, and `customValues` map covers all supported values.

**Review by andre-paraense — APPROVED** [review]

**Comment by andre-paraense** [comment]
@ookami-kb awesome, thanls for the contribution, will be merging this and releasing a new version as soon as possible, probably in the next couple of days! :) 

I plan to leave an explicit mention of your name in the version changelog, let me know if you are not comfortable with that.

**Comment by ookami-kb** [comment]
Cool, nice to hear that!

Yeah, I'm totally ok with it 👍
